docs/architecture.md

```diff
@@ -75,3 +116,114 @@ the decision.
 ### Attestation diagram
 
 ![attestation](images/Fobnail-flows-attestation.png)
+
+---
+
+## Keys and certificates
+
+### Platform Owner
+
+#### Knows in advance
+
+- Platform Owner key (signing and encryption)
+  - Fobnail and remote platform provisioning
+  - used for TLS and for signing certificates
+- Platform Owner certificate
+  - Fobnail and remote platform provisioning
+  - has to be able to provide whole CA chain
+  - both Fobnail Token and Platform must trust root CA in this chain
+- root CA certificate for EK certificate chain
+  - remote platform provisioning
+
+#### Receives
+
+- Fobnail Identity key
+  - Fobnail provisioning
+- Fobnail Encryption key
+  - Fobnail provisioning
+- EK certificate
+  - remote platform provisioning
+- AIK
+  - remote platform provisioning
+
+#### Produces
+
+- Fobnail Identity certificate
+  - Fobnail provisioning
+- Fobnail Encryption certificate
+  - Fobnail provisioning
+- AIK certificate
+  - remote platform provisioning
+  - simple signature instead of certificate may be sufficient
+
+### Fobnail Token
+
+#### Knows in advance
+
+- root CA certificate for EK certificate chain
+  - local platform provisioning
+  - possibly along with whole chain
```

Comment:
We could store only root CAs in Fobnail. Attester would have to determine chain (starting from EK), fetch all intermediates (certificate may URL where to certificates from) and pass them to Fobnail. Then Fobnail would keep only needed certificates.
Doing so ads significant complexity, but we could do it somewhere in future.

Reply:
This would be the most portable way. 3.2.13 https://trustedcomputinggroup.org/wp-content/uploads/TCG_IWG_EKCredentialProfile_v2p3_r2_pub.pdf says that it should be possible to obtain whole chain from AIA fields, at least if TPM is compatible with Windows.